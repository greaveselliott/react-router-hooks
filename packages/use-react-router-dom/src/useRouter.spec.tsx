import React, { FC } from "react";
import { HashRouter } from "react-router-dom";
import { RouteComponentProps } from "react-router-dom";
import "@testing-library/react/cleanup-after-each";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";

import { useRouter } from ".";

test("Calling useRouter to provides access to the routerContext.", () => {
  let routerContext: RouteComponentProps;

  const MockComponent: FC = () => {
    routerContext = useRouter();

    return <h1>Test</h1>;
  };

  render(
    <HashRouter>
      <MockComponent />
    </HashRouter>
  );

  expect(routerContext).toMatchSnapshot();
});
