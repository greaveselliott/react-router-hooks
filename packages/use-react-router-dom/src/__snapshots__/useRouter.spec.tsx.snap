// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Call useNavigate to configure URL template, then call the hook to populate the template. 1`] = `
Object {
  "history": Object {
    "action": "POP",
    "block": [Function],
    "createHref": [Function],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "location": Object {
    "hash": "",
    "pathname": "/",
    "search": "",
    "state": undefined,
  },
  "match": Object {
    "isExact": true,
    "params": Object {},
    "path": "/",
    "url": "/",
  },
  "staticContext": undefined,
}
`;

exports[`Calling useRouter to provides access to the routerContext. 1`] = `
Object {
  "history": Object {
    "action": "POP",
    "block": [Function],
    "createHref": [Function],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "location": Object {
    "hash": "",
    "pathname": "/",
    "search": "",
    "state": undefined,
  },
  "match": Object {
    "isExact": true,
    "params": Object {},
    "path": "/",
    "url": "/",
  },
  "staticContext": undefined,
}
`;
